### PR TITLE
Configplugin & registry 

### PR DESCRIPTION
**Pull request by bunsenstraat**
- 'config' is a plugin which can be called
- fixed async await issues when calling plugin
- use async in useeffect to get the variables
- replaced registry by TS singleton

### CODE REVIEW

File: apps/remix-ide/src/app/tabs/compile-tab.js [hunk 1]
```diff
@@ -144,17 +144,12 @@ class CompileTab extends CompilerApiMixin(ViewPlugin) { // implements ICompilerA
...
   }
 
-  getAppParameter (name) {
-    // first look in the URL params then in the local storage
-    const params = this.queryParams.get()
-    const param = params[name] ? params[name] : this.config.get(name)
-    if (param === 'true') return true
-    if (param === 'false') return false
-    return param
+  async getAppParameter (name) {
```
Comment (yann300) [inline apps/remix-ide/src/app/tabs/compile-tab.js:147]: are you certain that all the occurrences of `getAppParameter` use `await` now?

File: apps/remix-ide/src/app/editor/contextView.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ const css = require('./styles/contextView-styles')
 class ContextView {
   constructor (opts, localRegistry) {
     this._components = {}
-    this._components.registry = localRegistry || globalRegistry
+    this._components.registry = localRegistry || Registry.getInstance()
```
Comment (yann300) [inline apps/remix-ide/src/app/editor/contextView.js:18]: I guess you can now remove `localRegistry`  everywhere (this is basically the same as the global one)

File: apps/remix-ide/src/app/tabs/compile-tab.js (hunk 1 above)
Comment (bunsenstraat) [reply apps/remix-ide/src/app/tabs/compile-tab.js:147]: Can you find a place where it doesn't use it?

Comment (yann300) [reply apps/remix-ide/src/app/tabs/compile-tab.js:147]: I think this is fine.
btw maybe the`getAppParameter` from  `compiler.ts` needs to use the new config plugin then, or is this not part of this PR and can be done afterwards ?

Comment (bunsenstraat) [reply apps/remix-ide/src/app/tabs/compile-tab.js:147]: I saw that too, it confused me. The other one uses localstorage directly?

Comment (yann300) [reply apps/remix-ide/src/app/tabs/compile-tab.js:147]: yes, the impl of the one in the plugin, is too use url param if any and fallback to the localstorage.
I think it's fine if the version from `compiler.ts` uses that version

Comment (bunsenstraat) [reply apps/remix-ide/src/app/tabs/compile-tab.js:147]: That's better than I can change the api ts definition too.

**Review by yann300 — APPROVED** [review]

**Pull request merged**